website/blog/2024-10-23-the-new-architecture-is-here.md

```diff
@@ -0,0 +1,406 @@
+
+The overall strategy is to get your application running on the New Architecture without breaking existing code. You can then gradually migrate your app at your own pace. For new surfaces that have migrated all modules to the New Architecture, you can start using concurrent features immediately. For existing surfaces, you may need to address some issues and migrate modules before adding concurrent features.
+
+We've collaborated with the most popular React Native libraries to ensure support for the New Architecture. More than 850 libraries are already compatible, including all libraries with over 200K weekly downloads (~10% of downloaded libraries). You can check library compatibility with the New Architecture on the [reactnative.directory](https://reactnative.directory) website:
+
+![](/blog/assets/0.76-directory.png)
+
+For more details on upgrading, see [How to Upgrade](/blog/2024/10/23/the-new-architecture-is-here#how-to-upgrade) below.
+
+## New Features
+
+The New Architecture includes full support for React 18, concurrent features, and `useLayoutEffect` in React Native. For a full list of React 18 features, please see the [React 18 blog post](https://react.dev/blog/2021/12/17/react-conf-2021-recap#react-18-and-concurrent-features).
+
+### Transitions
+
+Transitions are a new concept in React 18 to distinguish between urgent and non-urgent updates.
+
+- **Urgent updates** reflect direct interaction, like typing and pressing.
+- **Transition updates** transition the UI from one view to another.
+
+Urgent updates need immediate response to match our intuitions about how physical objects behave. However, transitions are different because the user doesn’t expect to see every intermediate value on screen. In the New Architecture, React Native is able to support rendering urgent updates and transition updates separately.
+
+Typically, for the best user experience, a single user input should result in both an urgent update and a non-urgent one. Similar to ReactDOM, events like `press` or `change` are handled as urgent and rendered immediately. You can use the `startTransition` API inside an input event to inform React which updates are “transitions” and can be deferred to the background:
+
+```jsx
+import {startTransition} from 'react';
+
+// Urgent: Show the slider value
+setCount(input);
+
+// Mark any state updates inside as transitions
+startTransition(() => {
+  // Transition: Show the results
+  setNumberOfTiles(input);
+});
+```
+
+Separating urgent events from transitions allows for a more responsive user interface, and a more intuitive user experience.
+
+Here's a comparison of the old architecture without transitions (left) and the new architecture with transitions (right):
```

Comment:
Mentioning left/right is confusing on mobile since demos are displayed vertically. 

I'd suggest removing those and making it easier to scan figcaptions and figure out the before/after. For example starting with `Without transition - desc` and `With transition - desc`

Also, both transition/batching demos are not in a consistent order. Maybe always put the "old" behavior first/left, and the new behavior second/right?

This transition example is particularly confusing, because it looks to me the visual result of "without transition" is better. So much that I initially thought the gifs were interverted (and only the little transition isPending spinner confirmed me that it wasn't the case). 

I'd suggest adding a comment to help me figure out what I'm supposed to see here (same as the batching demo below) explaining why the transition looks worse but is actually good in practice. 

Maybe the no-transition demo could better show the problem in the first place, by show a lagging range slider where the thumb gets stuck and does not stay under the finger?